planck-cljs/src/planck/io.cljs

```diff
@@ -346,12 +346,67 @@
 (defn ^boolean directory?
   "Checks if dir is a directory."
   [dir]
-  (js/PLANCK_IS_DIRECTORY (:path (as-file dir))))
+  (boolean (js/PLANCK_IS_DIRECTORY (:path (as-file dir)))))
 
 (s/fdef directory?
   :args (s/cat :dir (s/or :string string? :file file?))
   :ret boolean?)
 
+(defn exists?
+  "Checks if f exists on disk."
+  [f]
+  (not (nil? (file-attributes f))))
+
+(s/fdef exists?
+  :args (s/cat :f (s/or :string string? :file file?))
+  :ret boolean?)
+
+(defn path-elements
+  "Returns the path elements of x as a sequence."
+  [x]
+  (if x
```

Comment:
Shouldn't need this guard for `x` as anything satisfying `string?` or `file?` is truth.

Reply:
Without that guard `path-elements` will return an empty sequence when given `nil`, whereas I prefer to return `nil` when handed `nil` in circumstances like this.

Reply:
Take `directory?`, for example, which doesn't need the `boolean` coercion. The spec says that its argument must satisfy `string?` or `file?`. If you pass `nil` with specs enabled, you can see that `nil` is an invalid argument:

```
$ plk
ClojureScript 1.10.516
cljs.user=> (require '[planck.io :as io] '[clojure.spec.test.alpha :as st])
nil
cljs.user=> (st/instrument `io/directory?)
[planck.io/directory?]
cljs.user=> (io/directory? nil)
Execution error - invalid arguments to planck.io/directory? at (<cljs repl>:1).
nil - failed: string? at: [:dir :string]
nil - failed: file? at: [:dir :file]
```